File: ios/Sasquatsh/Views/Shared/UserProfileSheet.swift
import SwiftUI

struct UserProfileSheet: View {
    @Environment(\.services) private var services
    @Environment(\.dismiss) private var dismiss

    let userId: String
    @State private var profile: PublicProfile?
    @State private var badges: [UserBadge] = []
    @State private var collection: [CollectionGame] = []
    @State private var isLoading = true

    var body: some View {
        NavigationStack {
            ScrollView {
                if isLoading {
                    VStack {
                        Spacer(minLength: 80)
                        ProgressView()
                        Spacer()
                    }
                } else if let profile {
                    VStack(spacing: 16) {
                        // Avatar + name
                        VStack(spacing: 10) {
                            UserAvatarView(url: profile.avatarUrl, name: profile.displayName, size: 72)

                            Text(profile.displayName ?? "Unknown")
                                .font(.md3HeadlineSmall)
                                .foregroundStyle(Color.md3OnSurface)

                            Text("@\(profile.username)")
                                .font(.md3BodyMedium)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                        }

                        // Location
                        if let city = profile.homeCity, let state = profile.homeState {
                            Label("\(city), \(state)", systemImage: "mappin.circle.fill")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                        }

                        // Bio
                        if let bio = profile.bio, !bio.isEmpty {
                            Text(bio)
                                .font(.md3BodyMedium)
                                .foregroundStyle(Color.md3OnSurface)
                                .multilineTextAlignment(.center)
                                .lineLimit(3)
                                .padding(.horizontal, 24)
                        }

                        // Pinned badges
                        let pinned = badges.filter(\.isPinned)
                        if !pinned.isEmpty {
                            VStack(spacing: 8) {
                                Text("Pinned Badges")
                                    .font(.md3LabelLarge)
                                    .foregroundStyle(Color.md3OnSurfaceVariant)

                                HStack(spacing: 12) {
                                    ForEach(pinned) { ub in
                                        VStack(spacing: 4) {
                                            badgeIcon(ub.badge)
                                            Text(ub.badge.name)
                                                .font(.md3BodySmall)
                                                .foregroundStyle(Color.md3OnSurface)
                                                .lineLimit(1)
                                        }
                                        .frame(width: 70)
                                    }
                                }
                            }
                            .padding(12)
                            .frame(maxWidth: .infinity)
                            .background(Color.md3Surface)
                            .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
                            .padding(.horizontal, 16)
                        }

                        // Badges summary
                        if !badges.isEmpty {
                            VStack(alignment: .leading, spacing: 8) {
                                HStack {
                                    Image(systemName: "trophy.fill")
                                        .foregroundStyle(Color.md3Tertiary)
                                    Text("\(badges.count) Badges Earned")
                                        .font(.md3TitleSmall)
                                        .foregroundStyle(Color.md3OnSurface)
                                }

                                // Show first 8 badges
                                LazyVGrid(columns: Array(repeating: GridItem(.flexible()), count: 4), spacing: 8) {
                                    ForEach(badges.prefix(8)) { ub in
                                        VStack(spacing: 2) {
                                            badgeIcon(ub.badge)
                                            Text(ub.badge.name)
                                                .font(.system(size: 9))
                                                .foregroundStyle(Color.md3OnSurfaceVariant)
                                                .lineLimit(1)
                                        }
                                    }
                                }

                                if badges.count > 8 {
                                    Text("+\(badges.count - 8) more")
                                        .font(.md3BodySmall)
                                        .foregroundStyle(Color.md3Primary)
                                        .frame(maxWidth: .infinity, alignment: .center)
                                }
                            }
                            .padding(12)
                            .frame(maxWidth: .infinity)
                            .background(Color.md3Surface)
                            .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
                            .padding(.horizontal, 16)
                        }

                        // Game collection
                        if !collection.isEmpty {
                            VStack(alignment: .leading, spacing: 8) {
                                HStack {
                                    Image(systemName: "dice")
                                        .foregroundStyle(Color.md3Primary)
                                    Text("Game Collection (\(collection.count))")
                                        .font(.md3TitleSmall)
                                        .foregroundStyle(Color.md3OnSurface)
                                }

                                ForEach(collection.prefix(5)) { game in
                                    HStack(spacing: 10) {
                                        if let urlStr = game.thumbnailUrl, let url = URL(string: urlStr) {
                                            AsyncImage(url: url) { image in
                                                image.resizable().aspectRatio(contentMode: .fill)
                                            } placeholder: {
                                                Color.md3SurfaceVariant
                                            }
                                            .frame(width: 32, height: 32)
                                            .clipShape(RoundedRectangle(cornerRadius: 4))
                                        }

                                        Text(game.gameName)
                                            .font(.md3BodySmall)
                                            .foregroundStyle(Color.md3OnSurface)
                                            .lineLimit(1)

                                        Spacer()

                                        if let min = game.minPlayers, let max = game.maxPlayers {
                                            Text(min == max ? "\(min)p" : "\(min)-\(max)p")
                                                .font(.md3BodySmall)
                                                .foregroundStyle(Color.md3OnSurfaceVariant)
                                        }
                                    }
                                }

                                if collection.count > 5 {
                                    NavigationLink {
                                        UserCollectionListView(games: collection, userName: profile.displayName ?? "User")
                                    } label: {
                                        Text("View all \(collection.count) games →")
                                            .font(.md3LabelLarge)
                                            .foregroundStyle(Color.md3Primary)
                                            .frame(maxWidth: .infinity, alignment: .center)
                                    }
                                }
                            }
                            .padding(12)
                            .frame(maxWidth: .infinity)
                            .background(Color.md3Surface)
                            .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
                            .padding(.horizontal, 16)
                        }

                        // Favorite games
                        if let favorites = profile.favoriteGames, !favorites.isEmpty {
                            VStack(alignment: .leading, spacing: 6) {
                                Text("Favorite Games")
                                    .font(.md3LabelLarge)
                                    .foregroundStyle(Color.md3OnSurfaceVariant)

                                FlowLayout(spacing: 6) {
                                    ForEach(favorites, id: \.self) { game in
                                        Text(game)
                                            .font(.md3BodySmall)
                                            .padding(.horizontal, 10)
                                            .padding(.vertical, 4)
                                            .background(Color.md3PrimaryContainer.opacity(0.5))
                                            .foregroundStyle(Color.md3OnPrimaryContainer)
                                            .clipShape(Capsule())
                                    }
                                }
                            }
                            .padding(12)
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .background(Color.md3Surface)
                            .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
                            .padding(.horizontal, 16)
                        }

                        Spacer(minLength: 16)
                    }
                    .padding(.top, 8)
                }
            }
            .background(Color.md3SurfaceContainer)
            .toolbar {
                ToolbarItem(placement: .confirmationAction) {
                    Button("Done") { dismiss() }
                }
            }
        }
        .task {
            await loadProfile()
        }
    }

    private func badgeIcon(_ badge: Badge) -> some View {
        ZStack {
            Circle()
                .fill(tierColor(badge.tier).opacity(0.15))
                .frame(width: 36, height: 36)
            Image(systemName: categoryIcon(badge.category))
                .font(.system(size: 16))
                .foregroundStyle(tierColor(badge.tier))
        }
    }

    private func loadProfile() async {
        isLoading = true
        do {
            async let profileTask = services.profile.getPublicProfile(userId: userId)
            async let badgesTask: () = loadBadges()
            async let collectionTask: () = loadCollection()

            profile = try await profileTask
            _ = await (badgesTask, collectionTask)
        } catch {
            print("Failed to load profile: \(error)")
        }
        isLoading = false
    }

    private func loadBadges() async {
        do {
            let response: BadgesResponse = try await services.api.get("badges", queryItems: [
                .init(name: "action", value: "user"),
                .init(name: "userId", value: userId)
            ], authenticated: false)
            badges = response.badges
        } catch {
            // Non-fatal
        }
    }

    private func loadCollection() async {
        do {
            struct CollectionWrapper: Decodable { let games: [CollectionGame]? }
            let response: CollectionWrapper = try await services.api.get("collections", queryItems: [
                .init(name: "userId", value: userId)
            ], authenticated: false)
            collection = response.games ?? []
        } catch {
            // Non-fatal — collection may be private
        }
    }

    private func tierColor(_ tier: String) -> Color {
        switch tier {
        case "bronze": return Color(red: 0.8, green: 0.5, blue: 0.2)
        case "silver": return Color(red: 0.63, green: 0.63, blue: 0.63)
        case "gold": return Color(red: 1.0, green: 0.84, blue: 0.0)
        case "platinum": return Color.md3Primary
        default: return Color.md3OnSurfaceVariant
        }
    }

    private func categoryIcon(_ category: String) -> String {
        switch category {
        case "hosting": return "house.fill"
        case "attendance": return "person.fill.checkmark"
        case "planning": return "calendar"
        case "social": return "person.3.fill"
        case "collection": return "dice"
        case "game_system": return "gamecontroller.fill"
        case "items": return "bag.fill"
        case "special": return "star.fill"
        default: return "questionmark.circle"
        }
    }
}

// Simple flow layout for tags
struct FlowLayout: Layout {
    var spacing: CGFloat = 6

    func sizeThatFits(proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) -> CGSize {
        let result = layout(proposal: proposal, subviews: subviews)
        return result.size
    }

    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
        let result = layout(proposal: proposal, subviews: subviews)
        for (index, position) in result.positions.enumerated() {
            subviews[index].place(at: CGPoint(x: bounds.minX + position.x, y: bounds.minY + position.y), proposal: .unspecified)
        }
    }

    private func layout(proposal: ProposedViewSize, subviews: Subviews) -> (size: CGSize, positions: [CGPoint]) {
        let maxWidth = proposal.width ?? .infinity
        var positions: [CGPoint] = []
        var x: CGFloat = 0
        var y: CGFloat = 0
        var rowHeight: CGFloat = 0

        for subview in subviews {
            let size = subview.sizeThatFits(.unspecified)
            if x + size.width > maxWidth && x > 0 {
                x = 0
                y += rowHeight + spacing
                rowHeight = 0
            }
            positions.append(CGPoint(x: x, y: y))
            rowHeight = max(rowHeight, size.height)
            x += size.width + spacing
        }

        return (CGSize(width: maxWidth, height: y + rowHeight), positions)
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/shared/UserProfileSheet.kt
package com.sasquatsh.app.views.shared

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.sasquatsh.app.models.Badge
import com.sasquatsh.app.models.CollectionGame
import com.sasquatsh.app.models.PublicProfile
import com.sasquatsh.app.models.UserBadge
import com.sasquatsh.app.services.BadgesService
import com.sasquatsh.app.services.CollectionsService
import com.sasquatsh.app.services.ProfileService
import kotlinx.coroutines.async
import kotlinx.coroutines.coroutineScope

/**
 * Reusable bottom sheet / full-screen composable showing another user's profile.
 * Port of the iOS UserProfileSheet.
 */
@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun UserProfileSheet(
    userId: String,
    profileService: ProfileService,
    badgesService: BadgesService,
    collectionsService: CollectionsService,
    onDismiss: () -> Unit,
    onViewAllGames: ((List<CollectionGame>, String) -> Unit)? = null
) {
    var profile by remember { mutableStateOf<PublicProfile?>(null) }
    var badges by remember { mutableStateOf<List<UserBadge>>(emptyList()) }
    var collection by remember { mutableStateOf<List<CollectionGame>>(emptyList()) }
    var isLoading by remember { mutableStateOf(true) }

    LaunchedEffect(userId) {
        isLoading = true
        try {
            coroutineScope {
                val profileDeferred = async {
                    try {
                        profileService.getPublicProfile(userId)
                    } catch (_: Exception) {
                        null
                    }
                }
                val badgesDeferred = async {
                    try {
                        badgesService.getUserBadges(userId)
                    } catch (_: Exception) {
                        emptyList<UserBadge>()
                    }
                }
                val collectionDeferred = async {
                    try {
                        collectionsService.getUserCollection(userId)
                    } catch (_: Exception) {
                        emptyList()
                    }
                }

                profile = profileDeferred.await()
                badges = badgesDeferred.await()
                collection = collectionDeferred.await()
            }
        } catch (_: Exception) {
            // Non-fatal
        }
        isLoading = false
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {},
                actions = {
                    TextButton(onClick = onDismiss) {
                        Text("Done")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.surfaceContainer
                )
            )
        },
        containerColor = MaterialTheme.colorScheme.surfaceContainer
    ) { innerPadding ->
        if (isLoading) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(innerPadding),
                contentAlignment = Alignment.Center
            ) {
                D20ProgressView(size = 32.dp)
            }
        } else if (profile != null) {
            val prof = profile!!
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(innerPadding)
                    .verticalScroll(rememberScrollState())
                    .padding(top = 8.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Avatar + name
                UserAvatarView(
                    url = prof.avatarUrl,
                    name = prof.displayName,
                    size = 72.dp
                )

                Spacer(modifier = Modifier.height(10.dp))

                Text(
                    text = prof.displayName ?: "Unknown",
                    style = MaterialTheme.typography.headlineSmall,
                    color = MaterialTheme.colorScheme.onSurface
                )

                Text(
                    text = "@${prof.username}",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )

                // Location
                if (prof.homeCity != null && prof.homeState != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Filled.LocationOn,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp),
                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text(
                            text = "${prof.homeCity}, ${prof.homeState}",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }

                // Bio
                if (!prof.bio.isNullOrBlank()) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = prof.bio,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurface,
                        textAlign = TextAlign.Center,
                        maxLines = 3,
                        overflow = TextOverflow.Ellipsis,
                        modifier = Modifier.padding(horizontal = 24.dp)
                    )
                }

                Spacer(modifier = Modifier.height(16.dp))

                // Pinned badges
                val pinned = badges.filter { it.isPinned }
                if (pinned.isNotEmpty()) {
                    SectionCard(modifier = Modifier.padding(horizontal = 16.dp)) {
                        Text(
                            text = "Pinned Badges",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            modifier = Modifier.align(Alignment.CenterHorizontally)
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.Center
                        ) {
                            pinned.forEach { ub ->
                                Column(
                                    modifier = Modifier.width(70.dp),
                                    horizontalAlignment = Alignment.CenterHorizontally
                                ) {
                                    BadgeIcon(badge = ub.badge, size = 36)
                                    Text(
                                        text = ub.badge.name,
                                        style = MaterialTheme.typography.bodySmall,
                                        color = MaterialTheme.colorScheme.onSurface,
                                        maxLines = 1,
                                        overflow = TextOverflow.Ellipsis
                                    )
                                }
                                Spacer(modifier = Modifier.width(12.dp))
                            }
                        }
                    }
                    Spacer(modifier = Modifier.height(12.dp))
                }

                // Badges summary
                if (badges.isNotEmpty()) {
                    SectionCard(modifier = Modifier.padding(horizontal = 16.dp)) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                imageVector = Icons.Filled.EmojiEvents,
                                contentDescription = null,
                                tint = MaterialTheme.colorScheme.tertiary,
                                modifier = Modifier.size(20.dp)
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = "${badges.size} Badges Earned",
                                style = MaterialTheme.typography.titleSmall,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                        }
                        Spacer(modifier = Modifier.height(8.dp))

                        // Show first 8 badges in a grid
                        val displayBadges = badges.take(8)
                        // Use a simple flow row instead of LazyVerticalGrid (inside scrollable)
                        FlowRow(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceEvenly,
                            maxItemsInEachRow = 4
                        ) {
                            displayBadges.forEach { ub ->
                                Column(
                                    modifier = Modifier
                                        .padding(4.dp)
                                        .width(70.dp),
                                    horizontalAlignment = Alignment.CenterHorizontally
                                ) {
                                    BadgeIcon(badge = ub.badge, size = 36)
                                    Text(
                                        text = ub.badge.name,
                                        fontSize = 9.sp,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                                        maxLines = 1,
                                        overflow = TextOverflow.Ellipsis
                                    )
                                }
                            }
                        }

                        if (badges.size > 8) {
                            Spacer(modifier = Modifier.height(4.dp))
                            Text(
                                text = "+${badges.size - 8} more",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.primary,
                                modifier = Modifier.fillMaxWidth(),
                                textAlign = TextAlign.Center
                            )
                        }
                    }
                    Spacer(modifier = Modifier.height(12.dp))
                }

                // Game collection
                if (collection.isNotEmpty()) {
                    SectionCard(modifier = Modifier.padding(horizontal = 16.dp)) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Text(
                                text = "\uD83C\uDFB2", // dice emoji
                                fontSize = 16.sp
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = "Game Collection (${collection.size})",
                                style = MaterialTheme.typography.titleSmall,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                        }
                        Spacer(modifier = Modifier.height(8.dp))

                        collection.take(5).forEach { game ->
                            Row(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(vertical = 2.dp),
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                if (game.thumbnailUrl != null) {
                                    AsyncImage(
                                        model = game.thumbnailUrl,
                                        contentDescription = game.gameName,
                                        contentScale = ContentScale.Crop,
                                        modifier = Modifier
                                            .size(32.dp)
                                            .clip(RoundedCornerShape(4.dp))
                                            .background(MaterialTheme.colorScheme.surfaceVariant)
                                    )
                                    Spacer(modifier = Modifier.width(10.dp))
                                }

                                Text(
                                    text = game.gameName,
                                    style = MaterialTheme.typography.bodySmall,
                                    color = MaterialTheme.colorScheme.onSurface,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis,
                                    modifier = Modifier.weight(1f)
                                )

                                val min = game.minPlayers
                                val max = game.maxPlayers
                                if (min != null && max != null) {
                                    Text(
                                        text = if (min == max) "${min}p" else "${min}-${max}p",
                                        style = MaterialTheme.typography.bodySmall,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                }
                            }
                        }

                        if (collection.size > 5 && onViewAllGames != null) {
                            Spacer(modifier = Modifier.height(4.dp))
                            TextButton(
                                onClick = {
                                    onViewAllGames(collection, prof.displayName ?: "User")
                                },
                                modifier = Modifier.fillMaxWidth()
                            ) {
                                Text(
                                    text = "View all ${collection.size} games \u2192",
                                    style = MaterialTheme.typography.labelLarge,
                                    color = MaterialTheme.colorScheme.primary
                                )
                            }
                        }
                    }
                    Spacer(modifier = Modifier.height(12.dp))
                }

                // Favorite games
                val favorites = prof.favoriteGames
                if (!favorites.isNullOrEmpty()) {
                    SectionCard(modifier = Modifier.padding(horizontal = 16.dp)) {
                        Text(
                            text = "Favorite Games",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.height(6.dp))
                        FlowRow(
                            horizontalArrangement = Arrangement.spacedBy(6.dp),
                            verticalArrangement = Arrangement.spacedBy(6.dp)
                        ) {
                            favorites.forEach { game ->
                                Text(
                                    text = game,
                                    style = MaterialTheme.typography.bodySmall,
                                    color = MaterialTheme.colorScheme.onPrimaryContainer,
                                    modifier = Modifier
                                        .clip(RoundedCornerShape(50))
                                        .background(
                                            MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.5f)
                                        )
                                        .padding(horizontal = 10.dp, vertical = 4.dp)
                                )
                            }
                        }
                    }
                    Spacer(modifier = Modifier.height(12.dp))
                }

                Spacer(modifier = Modifier.height(16.dp))
            }
        }
    }
}

/**
 * Reusable section card matching the iOS surface + rounded rect pattern.
 */
@Composable
private fun SectionCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(MaterialTheme.colorScheme.surface)
            .padding(12.dp),
        content = content
    )
}

/**
 * Badge icon with tier-colored circle background and category icon.
 */
@Composable
private fun BadgeIcon(badge: Badge, size: Int = 36) {
    val tColor = tierColor(badge.tier)
    Box(
        modifier = Modifier
            .size(size.dp)
            .clip(CircleShape)
            .background(tColor.copy(alpha = 0.15f)),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = categoryIconEmoji(badge.category),
            fontSize = (size / 2).sp
        )
    }
}

private fun tierColor(tier: String): Color {
    return when (tier) {
        "bronze" -> Color(0xFFCC8033)
        "silver" -> Color(0xFFA0A0A0)
        "gold" -> Color(0xFFFFD700)
        "platinum" -> Color(0xFF6366F1)
        else -> Color(0xFF9E9E9E)
    }
}

private fun categoryIconEmoji(category: String): String {
    return when (category) {
        "hosting" -> "\uD83C\uDFE0"        // house
        "attendance" -> "\u2714\uFE0F"      // check
        "planning" -> "\uD83D\uDCC5"        // calendar
        "social" -> "\uD83D\uDC65"          // people
        "collection" -> "\uD83C\uDFB2"      // dice
        "game_system" -> "\uD83C\uDFAE"     // game controller
        "items" -> "\uD83D\uDC5C"           // bag
        "special" -> "\u2B50"               // star
        else -> "\u2753"                    // question mark
    }
}
